### PR TITLE
OS10895818: F12 edge crash when debugging google.com

### PR DESCRIPTION
**Pull request by suwc**
For global function inside direct eval, FuncAllowsDirectSuper() looks up JavaScript caller through stackwalker to determine if the eval is called from a class constructor. Need to check if functionBody exists to cover the case where eval is called by F12 from inside a built-in function.

### CODE REVIEW

**Comment by suwc** [comment]
@curtisman @pleath @Microsoft/chakra-es review please

File: lib/Runtime/ByteCode/ByteCodeGenerator.cpp [hunk 1]
```diff
@@ -2585,7 +2585,7 @@ bool FuncAllowsDirectSuper(FuncInfo *funcInfo, ByteCodeGenerator *byteCodeGenera
     if (funcInfo->IsGlobalFunction() && ((byteCodeGenerator->GetFlags() & fscrEval) != 0))
     {
         Js::JavascriptFunction *caller = nullptr;
-        if (Js::JavascriptStackWalker::GetCaller(&caller, byteCodeGenerator->GetScriptContext()))
+        if (Js::JavascriptStackWalker::GetCaller(&caller, byteCodeGenerator->GetScriptContext()) && Js::ScriptFunction::Is(caller))
         {
```
Comment (agarwal-sandeep) [inline lib/Runtime/ByteCode/ByteCodeGenerator.cpp]: Can we add a test case in test\DebuggerCommon?

Comment (suwc) [inline lib/Runtime/ByteCode/ByteCodeGenerator.cpp]: I haven't been able to repro the issue with ch.exe/jshost.exe

**Comment by pleath** [comment]
LGTM. Thanks.

File: lib/Runtime/ByteCode/ByteCodeGenerator.cpp (hunk 1 above)
Comment (agarwal-sandeep) [inline lib/Runtime/ByteCode/ByteCodeGenerator.cpp]: I can help you with that

**Pull request merged**